packages/api/src/lib/structures/api/ApiRequest.ts

```diff
@@ -25,4 +27,32 @@ export class ApiRequest extends IncomingMessage {
 	 * - `AuthData`: The user is authorized.
 	 */
 	public auth?: AuthData | null;
+
+	/**
+	 * The router node that matched the request. The field indicates three
+	 * possible values:
+	 *
+	 * - `undefined`: The router handler has not been executed yet.
+	 * - `null`: The router handler has been executed, but no node matched the
+	 *   request.
+	 * - `RouterNode`: The router handler has been executed and a node matched
+	 *   the request.
+	 *
+	 * @since 7.0.0
+	 */
+	public routerNode?: RouterNode | null;
```

Comment:
Add in `undefined` too? Unless it's either set (null | RouterNode) or unset

Reply:
It's the latter, yes.